src/mono/mono/mini/mini-amd64.c

```diff
@@ -5645,6 +5652,14 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 			}
 
 			code = amd64_handle_varargs_call (cfg, code, call, TRUE);
+
+#ifdef MONO_ARCH_HAVE_SWIFTCALL
+			// Ideally, this would be in mono_arch_emit_prolog, but amd64_handle_varargs_call is required before the move to free RAX.
+			if (mono_method_signature_has_ext_callconv (cfg->method->signature, MONO_EXT_CALLCONV_SWIFTCALL) && 
+			    cfg->arch.cinfo->swift_indirect_result_index > -1) {
+				amd64_mov_reg_reg (code, AMD64_RAX, AMD64_R10, 8);
```

Comment:
I know that the PR description says that this PR includes changes for miniJIT and interpreter. However, do you know if these changes are sufficient for AOT as well?

Reply:
Good point, let's check it.

Reply:
Fixed, thanks.